samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/SavePartListener.java

```diff
@@ -321,7 +389,29 @@ public boolean validateItemsDrawn(SectionBean sectionBean){
 	     return false;
 
 	 }
-	
+
+     if (sectionBean.getType().equals(SectionDataIfc.FIXED_AND_RANDOM_DRAW_FROM_QUESTIONPOOL)) {
+		if ((StringUtils.isNotEmpty(numberSelectedFixed)) && (StringUtils.isNotEmpty(selectedPoolFixed))) {
+			int numberSelectedInt = Integer.parseInt(numberSelectedFixed);
+
+			// checking at least one fixed
+			if (numberSelectedInt == 0) {
+				err=ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.AuthorMessages","notfixedquestionselected_error");
+				context.addMessage(null,new FacesMessage(err));
+				return false;
+			}
+
+			int totalFixedPlusDrawInt = numberSelectedInt + numberDrawnInt;
+
+			// checking if selected same pool for fixed and drawn and there is not enough questions
+			if ((selectedPool.equals(selectedPoolFixed)) && (totalFixedPlusDrawInt > itemcount)) {
+				err=ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.AuthorMessages","notenough_error");
+				context.addMessage(null,new FacesMessage(err));
+				return false;
+			}
+		}
+    }
```

Comment:
Spacing here and at the initial if, I guess the file is a mess but maybe something that matches what comes next?

Reply:
Really hard to fix spacing on this piece of code...

Reply:
Yep, don't worry